hazelcast/src/main/java/com/hazelcast/internal/eviction/impl/evaluator/EvictionPolicyEvaluator.java

```diff
@@ -44,13 +42,15 @@ public EvictionPolicyComparator getEvictionPolicyComparator() {
     }
 
     /**
-     * The evaluate method implements the {@link com.hazelcast.config.EvictionPolicy} rule
-     * on the given input set of candidates.
+     * Selects the best candidate to be evicted.
+     * The definition of the best depends on configured eviction policy. (LRU, LFU, custom, etc)
+     *
+     * It returns <code>null</code> when there the input is empty.
      *
      * @param evictionCandidates Multiple {@link com.hazelcast.internal.eviction.EvictionCandidate} to be evicted
-     * @return multiple {@link com.hazelcast.internal.eviction.EvictionCandidate} these are available to be evicted
+     * @return a selected candidate to be evicted or null.
      */
-    public <C extends EvictionCandidate<A, E>> Iterable<C> evaluate(Iterable<C> evictionCandidates) {
+    public <C extends EvictionCandidate<A, E>> C evaluate(Iterable<C> evictionCandidates) {
```

Comment:
I though this whole method could be renamed to `selectBestCandidate()` ? @ahmetmircik: WDYT?

but perhaps not in this PR.

Reply:
seems doable